packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -131,11 +93,106 @@ function mapStandbyTransactionToFeedTokenTransaction(tx: StandbyTransaction): Fe
   }
 }
 
+function addFetchedTransactionFunction(
+  fetchedTransactions: TokenTransaction[],
+  setFetchedTransactions: any
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
 function TransactionFeed() {
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+
+  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  const [fetchMoreTransactions, setFetchMoreTransactions] = useState(false)
+
   const cachedTransactions = useSelector(transactionsSelector)
+  const [fetchedTransactions, setFetchedTransactions] = useState<TokenTransaction[]>([])
+  const addFetchedTransactions = addFetchedTransactionFunction(
```

Comment:
it seems like we are keeping `fetchedTransactions, setFetchedTransactions` in this component mainly to feed it into this hook. is it possible to change the interface of this hook to be something like `const [fetchedTransactions, addFetchedTransactions] = useFetchedTransactions()`? and this hook can manage the `fetchedTransactions` state variable

Reply:
Yes!! I like your idea! :)